src/ring_theory/int/basic.lean

```diff
@@ -341,3 +341,62 @@ instance decidable_int : decidable_rel (λ a b : ℤ, (multiplicity a b).dom) :=
 λ a b, decidable_of_iff _ finite_int_iff.symm
 
 end multiplicity
+
+lemma induction_on_primes [P: nat → Prop] (h₁: P 0) (h₂: P 1)
+  (h: ∀ p a: ℕ, nat.prime p → P a → P (p * a)): ∀ n: ℕ, P n :=
+begin
+  intro n,
+  apply unique_factorization_monoid.induction_on_prime,
+  exact h₁,
+  { intros n h,
+    rw nat.is_unit_iff.1 h,
+    exact h₂, },
+  { intros a p _ hp ha,
+    exact h p a (nat.prime_iff_prime.2 hp) ha, },
+end
+
+lemma int.associated_nat_abs (k: ℤ): associated k k.nat_abs :=
+begin
+  apply associated.symm,
+  unfold associated,
+  cases int.nat_abs_eq k with h h,
+  use 1, simp, exact h.symm,
+  use -1, simp, exact h.symm,
```

Comment:
We tend to avoid `simp` unless it's the last tactic in a block. `simpa using h.symm` might work if I recall the syntax correctly.

Reply:
Your other comment simply removes these lines, but thanks for the advice :)